zulipterminal/ui_tools/boxes.py

```diff
@@ -107,8 +108,9 @@ def generic_autocomplete(self, text: str, state: int) -> Optional[str]:
         else:
             return text
 
+        reduced_typeahead = typeahead[:num_suggestions]
```

Comment:
Writing tests is a little difficult, so I've let this be since its a small change.